src/NuGet.Core/NuGet.ProjectModel/CacheFileFormat.cs

```diff
@@ -2,52 +2,61 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
-using System.Collections.Generic;
 using System.Globalization;
 using System.IO;
-using Newtonsoft.Json;
-using Newtonsoft.Json.Linq;
+using System.Text.Encodings.Web;
+using System.Text.Json;
+using System.Text.Json.Serialization;
 using NuGet.Common;
 
 namespace NuGet.ProjectModel
 {
     public static class CacheFileFormat
     {
-        private const string VersionProperty = "version";
-        private const string DGSpecHashProperty = "dgSpecHash";
-        private const string SuccessProperty = "success";
-        private const string ExpectedPackageFilesProperty = "expectedPackageFiles";
-        private const string ProjectFilePathProperty = "projectFilePath";
-
-        public static CacheFile Read(Stream stream, ILogger log, string path)
+        private static JsonSerializerOptions SerializerOptions = new JsonSerializerOptions
         {
-            if (stream == null) throw new ArgumentNullException(nameof(stream));
-            if (log == null) throw new ArgumentNullException(nameof(log));
-            if (path == null) throw new ArgumentNullException(nameof(path));
+            WriteIndented = true,
+            PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+            Encoder = JavaScriptEncoder.UnsafeRelaxedJsonEscaping,
+            DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull,
+            Converters = { new AssetsLogMessageConverter() }
```

Comment:
```suggestion
            NumberHandling = JsonNumberHandling.AllowReadingFromString,
            Converters = { new AssetsLogMessageConverter() }
```

this will match netwonsoft behavior, then both `"version": "1"` and `"version": 1` will desrialize to int property as before